src/OrleansHost/WindowsServerHost.cs

```diff
@@ -114,7 +108,8 @@ private int RunImpl(CancellationToken? cancellationToken = null)
         public bool ParseArguments(string[] args)
         {
             string siloName = Dns.GetHostName(); // Default to machine name
-            SiloHost = new SiloHost(siloName);
+            var builder = new SiloHostBuilder();
+            SiloHost = builder.Build(siloName);
 
```

Comment:
This can be moved.  I was just getting it work.  However, I wouldn't have separate publically callable initialization methods.  The builder should take care of every option before creation.  Once something is created, it is fully initialized and ready to be used.